stripe-node/app.js

```diff
@@ -52,31 +50,53 @@ app.post('/charge/angular', (req, res) => {
     source: req.body.token.id,
     shipping: address,
     name: req.body.token.name,
-  })
-  // Add charge to customer
-    .then(customer => stripe.charges.create({
+  }, (err, customers) => {
+    stripe.charges.create({
       amount: req.body.amount,
       currency: req.body.currency,
       description: 'Donation to Keyman',
-      customer: customer.id,
+      customer: customers.id,
       receipt_email: req.body.token.email,
-    }));
-// Successfull api call
-  // .then(res => {
-  //   res.status(200)
-  //     .json({
-  //       status: '200',
-  //       message: res,
-  //     });
-  // })
-  // Handle api failure
-  // .catch(res => {
-  //   res.status(500)
-  //      .json({
-  //        status: '500',
-  //        message: res,
-  //      });
-  // })
+    }, (err, charges) => {
+      // Handle errors
+      console.log(err.type);
+      if (err !== null) {
+        switch (err.type) {
+          case 'StripeCardError':
+            // A declined card error
+            err.message = "Your card's expiration year is invalid.";
+            res.send(err.message);
```

Comment:
I would prefer to see error messages sent in some structured form (JSON?) from an API, rather than as plain text. Also, do you just send a HTTP/200 back to the website? Don't we need some sort of extra data, e.g. receipt #, etc?

Reply:
The json response is definitely doable. I'll see about working within the solution provided by Stripe (https://stripe.com/docs/api/errors/handling) and sending back json. So as far as the receipt is concerned, that depends on how we are sending receipts. Stripe integrates and sends receipts on behalf of the data collected (which you should have a copy of that example receipt).  If we are sending the receipts manually, I agree this data would need to be sent either back to Angular for download.

Reply:
Something needs to be sent back to the user. If they mistype their email address, etc, then some reference number is important.

Reply:
Can you clarify on what you mean by "if they mistype their email address, etc"

Reply:
Okay. Imagine the user goes through the process of making a donation. But instead of typing `marc@example.com`, they type `mar@example.com`. This would mean they would never receive the emailed receipt. What would they see on the screen in that instance with your current design? Would they see any receipt number, or other reference number that they would be able to contact us with -- seeing as we would not be able to contact them?

Reply:
Makes sense. I will work on getting this added (sending back receipt number).

Reply:
^^ *in-the-next-pr*